File: src/errors/ErrorHandler.java
import com.fasterxml.jackson.databind.ObjectMapper;
import java.io.File;
import java.io.IOException;
import java.util.Map;

public class ErrorHandler {
    private static Map<String, Map<String, Object>> errorCodes;

    static {
        try {
            ObjectMapper objectMapper = new ObjectMapper();
            errorCodes = objectMapper.readValue(new File("src/errors/errorCodes.json"), Map.class);
        } catch (IOException e) {
            throw new RuntimeException("Failed to load error mappings", e);
        }
    }

    public static Map<String, Object> getError(String errorKey) {
        return errorCodes.getOrDefault(errorKey, errorCodes.get("SYSTEM_INTERNAL_ERROR"));
    }
}

File: src/errors/errorHandler.js
const fs = require('fs');
const path = require('path');

// Load error mappings
const errorCodes = JSON.parse(fs.readFileSync(path.join(__dirname, 'errorCodes.json'), 'utf-8'));

class ErrorHandler {
    static getError(errorKey) {
        return errorCodes[errorKey] || errorCodes["SYSTEM_INTERNAL_ERROR"];
    }

    static errorResponse(res, errorKey) {
        const error = this.getError(errorKey);
        return res.status(error.httpStatus).json({
            status: "FAILED",
            error: {
                code: error.code,
                message: error.message
            }
        });
    }
}

const ErrorKeys = Object.keys(errorCodes).reduce((acc, key) => {
    acc[key] = key;
    return acc;
}, {});

module.exports = { ErrorHandler, ErrorKeys };
